src/dtype.rs

```diff
@@ -670,7 +711,7 @@ impl Sealed for Bound<'_, PyArrayDescr> {}
 ///
 /// [enumerated-types]: https://numpy.org/doc/stable/reference/c-api/dtype.html#enumerated-types
 /// [data-models]: https://en.wikipedia.org/wiki/64-bit_computing#64-bit_data_models
-pub unsafe trait Element: Clone + Send {
+pub unsafe trait Element: PyClone + Send {
```

Comment:
I wonder, rather than introducing a separate `PyClone` trait, which is specifically for the `Element` implementation, does it make sense to merge the `PyClone` methods into `Element`?

That probably allows for replacement of `IS_COPY` with a bunch of specialised methods for copying array elements back and forth.

Reply:
I think I agree with this suggestion; the main benefit of keeping it separate would be if we moved `PyClone` to pyo3 directly (perhaps as `doc(hidden)`) where we would have more freedom to provide the impls for both `Py<T>` and all `T: Clone`. But there are a few hurdles to doing so (e.g. needing to add an optional `ndarray` dependency to pyo3, and to wait for another release) that probably aren't worth jumping over.

Reply:
Fixed. One note is that I renamed the migrated `py_clone` method to `clone_ref` since the original name only really made sense in the context of a dedicated `PyClone` trait.